pulp_ansible/app/tasks/deletion.py

```diff
@@ -0,0 +1,78 @@
+"""
+This module includes tasks related to deleting Content.
+
+You can remove Content from a Repository by making a new RepositoryVersion
+without the Content. If an API endpoint uses a Distribution which points to
+the latest_version of the Repository then the Content is unavailable,
+however it is not deleted.
+
+Content can only be deleted if it exists in no RepositoryVersion.
+
+Content cannot be removed from a RepositoryVersion since it is immutable.
+
+Pulp's orphan_cleanup task deletes any Content not part of a RepositoryVersion.
+"""
+
+import logging
+
+from pulp_ansible.app.models import Collection, CollectionVersion
+from pulpcore.plugin.tasking import add_and_remove, orphan_cleanup
+
+log = logging.getLogger(__name__)
+
+
+def _cleanup_old_versions(repo):
+    """Delete all the old versions of the given repository."""
+    for version in repo.versions.complete().order_by("-number")[1:]:
+        version.delete()
+
+
+def _remove_collection_version_from_repos(collection_version):
+    """Remove CollectionVersion from latest RepositoryVersion of each repo."""
+    for repo in collection_version.repositories.all():
+        add_and_remove(repo.pk, add_content_units=[], remove_content_units=[collection_version.pk])
+        _cleanup_old_versions(repo)
+    collection_version.save()
```

Comment:
Content is immutable. We should not attempt to save it.

Reply:
@mdellweg can't we make `.save` on content to raise error when object already exists?

Reply:
Why should it raise an error? It would just see if ther is the need for a update in the database. I would look if it had a PK. In this case we are about to delete that thing. We can be pretty sure it exists. No?

Reply:
you said it is immutable, if it is immutable I should not be allowed to call mutating methods.